### PR TITLE
New core setting for server root

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

**Comment by jbeezley** [comment]
Does this overlap the intent of the proxy settings that cherrypy uses?  For example in the [deployment docs](http://girder.readthedocs.io/en/latest/deploy.html#girder-settings), the api root is overridden in the settings.

For a more pathological question, wouldn't this make it impossible to host two girder instances from the same database or to proxy a single girder instance to multiple mount points?

**Comment by zachmullen** [comment]
> Does this overlap the intent of the proxy settings that cherrypy uses? 

This is for special cases like @manthey 's where the girder instance may have a different URL on one machine than another (e.g. a VM). This shouldn't interfere with the proxy settings.

> wouldn't this make it impossible to host two girder instances from the same database or to proxy a single girder instance to multiple mount points?

If this setting is used, yes. However this is a configurable behavior and off by default; if not set, the behavior will be the same as before, which would enable those cases.

**Comment by zachmullen** [comment]
@manthey this is now ready for review

**Comment by zachmullen** [comment]
Actually sorry, I need to add front-end for it still

**Comment by zachmullen** [comment]
Ok, actually ready now.

**Comment by manthey** [comment]
Although this works for girder_worker, it breaks Swagger.  Would it be better to have this as a setting for the remote worker rather than core?  Or could we have Swagger not use the override logic?

**Comment by zachmullen** [comment]
Ah right, because in your case there is no one address where girder is visible to all the relevant machines... In that case, it makes sense to have this in the worker plugin as a setting.

**Comment by zachmullen** [comment]
I'll keep this setting too for whatever exotic use case my emerge. It's under advanced settings anyway, so it's probably not harmful.

File: girder/conf/girder.dist.cfg [hunk 1]
```diff
@@ -18,7 +18,7 @@ replica_set = None
 [server]
 # Set to "production" or "development"
 mode = "development"
-api_root = "api/v1"
+api_root = "/api/v1"
```
Comment (manthey) [inline girder/conf/girder.dist.cfg]: Since you lstrip the `/`, what is the virtue of adding it here?

**Comment by zachmullen** [comment]
I was planning to add the worker setting to this branch, would you prefer it on a separate PR?

**Comment by manthey** [comment]
Add away.  I'll re-review it then.

**Comment by manthey** [comment]
One minor point of confusion:  In the core setting, a value like `http://127.0.0.1:8080` automatically gets `/api/v1` appended to it (or whatever the config setting for api_url is).  In the remote worker setting, the value is the entire string (e.g., `http://127.0.0.1:8080/api/v1`).  We should document this somewhere.

**Pull request merged**